### ISSUE TITLE
Remote triggering of "local training"

### ISSUE THREAD

**Issue by oleschwen** [KatherLab/MediSwarm#208]
Think about whether there is a way to use the admin node/swarm clients to run a training equivalent to a local training.

**Comment by Ultimate-Storm** [comment]
## Analysis: Remote triggering of "local training"

Investigated whether an admin could submit an NVFlare job to trigger a local training equivalent on a running swarm client.

### Current architecture

There are two fundamentally different execution paths in `docker.sh`:

| | `--local_training` | `--start_client` (swarm) |
|---|---|---|
| **Docker lifecycle** | `docker run --rm` — runs, trains, exits | `docker run -d` — long-running daemon |
| **What runs** | `main.py` directly | `start.sh` → NVFlare client process |
| **NVFlare involved?** | **No** — `main.py` checks `TRAINING_MODE=local_training` and calls `validate_and_train()` directly | **Yes** — `flare.patch(trainer)`, then loops on `flare.receive()` waiting for jobs from the server |
| **Parameters from** | Container env vars (`MODEL_NAME`, `NUM_EPOCHS`, etc.) | Job config (`config_fed_client.conf`) + env vars |

### Why this is not a good fit

1. **NVFlare jobs are federated by design.** When a job is submitted via the admin API, the `SwarmServerController` orchestrates it across *all* participating clients simultaneously (aggregation rounds, model sharing, etc.). There's no built-in "run this job on only one client in isolation without aggregation" mode.

2. **`main.py` has a hard fork.** The training code (`main.py`) explicitly branches: swarm mode patches NVFlare into the trainer and loops on `flare.receive()`, while local mode just calls `validate_and_train()` once and exits. A submitted NVFlare job would always go through the swarm path.

3. **Training params aren't part of the NVFlare job config.** `config_fed_client.conf` defines the executor, aggregator, pipes — framework-level wiring. The training parameters (`MODEL_NAME`, `NUM_EPOCHS`, `CONFIG`) live in the container's environment. NVFlare's `app_config` field could theoretically pass args to `main.py`, but the code doesn't parse them that way.

4. **Would require a fundamentally different job type.** You'd need a new NVFlare workflow/controller (server-side) that dispatches a "local-only" task to a specific client, a new training mode in `main.py`, and a mechanism to pass training params through the NVFlare job config instead of env vars.

### Recommendation

The complexity is high and it fights NVFlare's design. "Local training" is inherently a single-participant, pre-swarm testing activity, so it makes sense for the participant to trigger it locally.

The original motivation (env vars getting lost with `sudo`) was addressed in #230 by adding CLI flags (`--model_name`, `--num_epochs`, `--config`) that survive `sudo`. Participants can now simply run:

```bash
sudo ./docker.sh --data_dir $DATADIR --scratch_dir $SCRATCHDIR --GPU device=0 --model_name challenge_3agaldran --local_training
```

Suggest closing this as **won't fix** — the use case is better served by the local CLI approach from #230.

**Comment by Ultimate-Storm** [comment]
## Updated plan (2026-07-02): v1.5.0 tooling opens a much cheaper path than the April estimate

The April analysis above still holds — a *literal* remote `--local_training` (bypassing NVFlare) fights the architecture, and the original motivation (env vars lost under `sudo`) was solved by #230's CLI flags. **If** remote triggering is still wanted, though, v1.5.0 makes a *training-equivalent* implementable without any new controller:

### Approach: a 1-client swarm job ≈ remotely-triggered local training

Aggregation over a single client is the identity function, so a swarm job scoped to one site trains on exactly that site's data — functionally a local training, triggered from the admin console. Since v1.5.0, `kit_admin_tools/prepare_odelia_job.sh` already patches quorum/rounds at prep time, which is most of the machinery:

1. **Validate a 1-client CCWF run** in the simulator/POC first: `--configure-min-clients 1 --min-clients 1 --min-responses 1 --num-rounds 1` and the target site as the starting client. (2-client runs are well-tested; 1-client is expected to work but unverified.)
2. **Add a `--site <SITE_1>` flag to `prepare_odelia_job.sh`** that (a) restricts `meta.json`'s `deploy_map` to server + that site, and (b) sets the CCWF controller's client list (`clients` / starting client) to it. ~30 lines in the existing patch step.
3. **Epochs mapping:** one round of `MAX_EPOCHS` = the local-training epoch budget (`--num-rounds 1`; `MAX_EPOCHS` rides with the client env as today).
4. **Docs:** short section in `assets/readme/README.operator.md` ("run a single-site training remotely").

### Caveats (why it's *equivalent*, not identical)

- NVFlare still wraps training (persistor, `/scratch` mirror, job workspace layout) — results land in the site's job workspace like any swarm run, not in the `--local_training` output layout.
- The client must be up as a swarm client (daemon mode), not idle.
- Params come from job config + kit env, not ad-hoc CLI flags.

### Decision for maintainers

- **Option A — close as won't fix** (April recommendation): participants trigger local training via #230's CLI flags; no new surface to maintain.
- **Option B — implement the 1-client-job path above**: S–M effort, useful for centrally-driven per-site baselines (e.g. #275-style comparisons at production sites) without asking site operators to run anything.

If B is chosen, step 1 (simulator validation) is the go/no-go gate before touching `prepare_odelia_job.sh`.
